File: 78/subsets.py
from typing import List
class Solution:
    def genSubsets(self,nums:List[int])->List[List[int]]:
        res=[]
        def backtrack(start,arr):
            res.append(arr[:])
            for i in range(start,len(nums)):
                arr.append(nums[i])
                backtrack(i+1,arr)
                arr.pop()

        backtrack(0,[])
        return res
sol=Solution()
nums=[1,2,3]
print(sol.genSubsets(nums))

File: 78/subsets.cpp
#include<iostream>
#include<vector>
using namespace std;
void backtrack(vector<int>& nums,vector<vector<int>>& res,vector<int>& curr,int ind)
{
    res.push_back(curr);
    for(int i=ind;i<nums.size();i++)
    {
        curr.push_back(nums[i]);
        backtrack(nums,res,curr,i+1);
        curr.pop_back();
    }
}
vector<vector<int>> findSubsets(vector<int>& nums)
{
    vector<vector<int>> res;
    vector<int> curr={};
    backtrack(nums,res,curr,0);
    return res;
}
int main()
{
    vector<int> nums={1,2,3};
    vector<vector<int>> res=findSubsets(nums);
    return 0;
}

